### PR TITLE
First entry in the FAQ: Vectors vs Tensors?

### PR DESCRIPTION
**Pull request by joker-eph**

### CODE REVIEW

File: website/content/getting_started/Faq.md [hunk 1]
```diff
@@ -5,4 +5,8 @@ menu: "main"
 weight: 10
 ---
 
-TODO
+## What is the difference between the Tensor and Vector types?
```
Comment (silvasean) [inline website/content/getting_started/Faq.md:8]: allowed element types are different too.

File: website/content/getting_started/Faq.md [hunk 2]
```diff
@@ -5,4 +5,9 @@ menu: "main"
...
+
+1) Conceptual: vectors are meant to and occur in lower level dialects - often where you expect hardware to have registers of that size. Tensors model higher-level "closer to the source" abstract representation. This is reflected in the abstraction modeled by the operations from the [`vector` dialect](https://mlir.llvm.org/docs/Dialects/Vector/), while Tensors would be more naturally present in the operations of the [`linalg dialect](https://mlir.llvm.org/docs/Dialects/Linalg/).
+2) Tensors can be dynamically shaped, unranked, or have 0 dimensions ; but Vectors can be.
```
Comment (bondhugula) [inline website/content/getting_started/Faq.md:11]: can be -> can't

Comment (joker-eph) [reply website/content/getting_started/Faq.md:11]: Thanks, fixed!

File: website/content/getting_started/Faq.md [hunk 3]
```diff
@@ -5,4 +5,9 @@ menu: "main"
 weight: 10
 ---
 
-TODO
+## What is the difference between the Tensor and Vector types?
+
+1) Conceptual: vectors are meant to and occur in lower level dialects - often where you expect hardware to have registers of that size. Tensors model higher-level "closer to the source" abstract representation. This is reflected in the abstraction modeled by the operations from the [`vector` dialect](https://mlir.llvm.org/docs/Dialects/Vector/), while Tensors would be more naturally present in the operations of the [`linalg dialect](https://mlir.llvm.org/docs/Dialects/Linalg/).
```
Comment (jpienaar) [inline website/content/getting_started/Faq.md:10]: "to and occur" ?

File: website/content/getting_started/Faq.md [hunk 4]
```diff
@@ -5,4 +5,9 @@ menu: "main"
...
+2) Tensors can be dynamically shaped, unranked, or have 0 dimensions ; but Vectors can be.
+3) You can have a memref (a buffer in memory) containing Vectors but you can't have a memref of a tensor type.
+4) The set of allowed element types is different: the Tensor type isn't limited while Vector is limited to float and integer types.
```
Comment (jpienaar) [inline website/content/getting_started/Faq.md:13]: Part of this also goes with tensors being abstract values that need not have any in memory backing. E.g., a tensor need not have any memory backing or fixed size elements. You can have a tensor of python programs as it is just an abstract value.

**Pull request merged**